File: src/main/java/com/patrickgrimard/Application.java
package com.patrickgrimard;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import java.util.Arrays;
import java.util.Collection;

@SpringBootApplication
@RestController
public class Application {

	public static void main(String[] args) {
		SpringApplication.run(Application.class, args);
	}

    @RequestMapping(value = "/items", method = RequestMethod.GET, produces = {MediaType.APPLICATION_JSON_VALUE})
    public Collection<String> items() {
        return Arrays.asList("item 1", "item 2", "item 3");
    }
}

File: src/main/java/com/patrickgrimard/Application.kt
package com.patrickgrimard

import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.context.annotation.Bean

/**
 *
 *
 * Created on 2016-03-24
 *
 * @author Patrick
 */
@SpringBootApplication
open class Application {

    @Bean
    open fun messageService() : MessageService {
        return KotlinMessageService()
    }
}

fun main(args: Array<String>) {
    SpringApplication.run(Application::class.java, *args)
}